### ISSUE TITLE
Execution error when transforming xml with an xml-model processing instruction

### ISSUE THREAD

**Issue by egli** [eerohele/sigel#4]
`xslt/transform` fails with 

```
Execution error (IllegalArgumentException) at sigel.protocols/eval52799$fn$G (protocols.clj:12).
No implementation of method: :build of protocol: #'sigel.protocols/XMLSource found for class: net.sf.saxon.s9api.XdmValue
```
if I invoke it on the following xml
``` xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://xmlp-test/schema/minimalSchema.rng"?>
<dtbook version="2005-3" xml:lang="de" xmlns="http://www.daisy.org/z3986/2005/dtbook/">
</dtbook>
```
The problem is the `<?xml-model href="http://xmlp-test/schema/minimalSchema.rng"?>` part which seems to be legal according to the [standard](https://www.w3.org/TR/xml-model/#the-xml-model-processing-instruction).

I transform with the following identity xsl:
``` xsl
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:xs="http://www.w3.org/2001/XMLSchema" exclude-result-prefixes="#all" version="2.0">

    <xsl:template match="@*|node()">
        <xsl:copy>
            <xsl:apply-templates select="@*|node()"/>
        </xsl:copy>
    </xsl:template>

</xsl:stylesheet>
```
The Clojure code to reproduce the problem is as follows:
``` clojure
(xslt/transform
 [(xslt/compile-xslt "identity.xsl")
  (xslt/compile-xslt "identity.xsl")]
 (io/file "bar.xml"))
```

**Comment by eerohele** [comment]
Thanks for the good bug report!

It turns out the issue is actually not related to `xml-model`. There was a bug in `sigel.xslt.core/pipeline` that caused the transformation to fail when any step in the pipeline yielded a result that appeared to have more than one root element (as was the case in your example).

I just cut a new release (v1.0.1). Can you give it a go and let me know whether it fixes the issue for you?

**Comment by egli** [comment]
Wow, that was fast! Works like a charm, thank you very much.

**Comment by egli** [comment]
Hi @eerohele sorry to bother you agin. Could it be that you forgot to push the 1.0.1 tag to github? It looks like that might be the reason why the import at [cljdoc](https://cljdoc.org/d/me.flowthing/sigel/1.0.1) failed.

**Comment by eerohele** [comment]
@egli I did forget. Too many manual steps in the deployment process...

I pushed the tag just now. Thanks for reminding me!

**Comment by eerohele** [comment]
That didn't seem to fix the cljdoc issue, though. Not sure how to fix it, but there's no difference between 1.0.0 and 1.0.1 as far as the API docs go.
